osi_logicallane.proto

```diff
@@ -0,0 +1,502 @@
+// \c #TYPE_INTERSECTION.
+//
+// If OSI is generated from OpenDRIVE, then LogicalLanes map directly to
+// OpenDRIVE lanes. However, it is allowed to merge multiple consecutive
+// OpenDRIVE lanes with the same type into one OSI LogicalLane: if an OpenDRIVE
+// lane has a single successor, which has the same lane type, and this
+// successor has only one predecessor (so no lane merging or splitting) then
+// the two lanes may be presented as one continuous LogicalLane. This may be
+// done recursively.
+//
+// The reference line pointed to by reference_line_id defines an ST coordinate
+// system for the lane. This ST coordinate system is used to describe positions
+// on the lane.
+//
+message LogicalLane
+{
+    // The ID of the logical lane.
+    //
+    // \note Note ID is global unique.
+    //
+    // \rules
+    // is_globally_unique
+    // \endrules
+    //
+    optional Identifier id = 1;
+
+    // The type of the logical lane.
+    //
+    optional Lane.Classification.Subtype type = 2;
+
+    // Optional external reference to the lane source.
+    //
+    // The external reference points to the source of the lane, if it is derived
+    // from one or more objects or external references.
+    //
+    // For example, to reference a lane defined in an OpenDRIVE map
+    // the items should be set as follows:
+    // * reference = URI to map, can remain empty if identical with definiton
+    //               in \c GroundTruth::map_reference
+    // * type = "net.asam.opendrive"
+    // * identifier[0] = id of t_road
+    // * identifier[1] = s of t_road_lanes_laneSection
+    // * identifier[2] = id of t_road_lanes_laneSection_left_lane,
+    //                         t_road_lanes_laneSection_right_lane
+    //
+    // \note For non-ASAM Standards, it is implementation-specific how
+    //       source_reference is resolved.
+    //
+    // \note The value has to be repeated, because one lane segment may be
+    //       derived from more than one origin segment. Multiple sources
+    //       may be added as reference as well, for example, a map and sensors.
+    //
+    repeated ExternalReference source_reference = 3;
+
+    // Reference to the physical lane this logical lane belongs to.
+    //
+    // This makes it possible to get detailed information on the physical lane
+    // properties, e.g. the visual colors of the boundaries, the road
+    // condition, etc..
+    //
+    // TODO Should this be a repeated field? Imagine the following scenario:
+    // ---------------------------------
+    //  l1                       l2
+    // ----------    l3        ---------
+    //  l4                       l5
+    // ---------------------------------
+    //
+    // In this case, we have five physical lanes: l1, l2, l4 and l5 where a
+    // lane marking is present. And l3 is one lane covering the whole road
+    // (because no road marking is present).
+    //
+    // Now there are two basic possibilities to define the logical lanes:
```

Comment:
We (Esther and I) would prefer option 1 with the two logical lanes for this example.

Reply:
Changed as per your preference :)